src/gfn/states.py

```diff
@@ -980,29 +987,23 @@ def backward_masks(self) -> TensorDict:
                 if len(edge_idx.shape) == 2:
                     edge_idx = edge_idx.sum(0).bool()
 
-                # Allow the removal of this edge
-                edge_masks[i, edge_idx] = True
+                edge_masks[i, : len(edge_idx)][edge_idx] = True
 
+        node_class_masks = node_class_masks.view(
+            *self.batch_shape, self.num_node_classes
+        )
         edge_masks = edge_masks.view(*self.batch_shape, max_possible_edges)
 
         # There are 3 action types: ADD_NODE, ADD_EDGE, EXIT
         action_type = torch.zeros(
             *self.batch_shape, 3, dtype=torch.bool, device=self.device
         )
-        action_type[..., GraphActionType.ADD_NODE] = torch.tensor(
-            [graph.x is not None and graph.x.size(0) > 0 for graph in self.data.flat],
-            device=self.device,
-        ).view(self.batch_shape)
+        action_type[..., GraphActionType.ADD_NODE] = torch.any(node_class_masks, dim=-1)
```

Comment:
this will be called node index instead,  as the backward policy will sample index instead of class